src/main/java/org/opensearch/neuralsearch/search/query/HybridCollectorManager.java

```diff
@@ -46,12 +51,13 @@ public abstract class HybridCollectorManager implements CollectorManager<Collect
 
     private final int numHits;
     private final HitsThresholdChecker hitsThresholdChecker;
-    private final boolean isSingleShard;
     private final int trackTotalHitsUpTo;
     private final SortAndFormats sortAndFormats;
     @Nullable
     private final Weight filterWeight;
     private static final float boost_factor = 1f;
+    private final ScoreDocsMerger<ScoreDoc> scoreDocsMerger = new ScoreDocsMerger<>();
```

Comment:
Can this initialization be moved to constructor and initialize only if concurrent segment search is enabled?

Reply:
what benefit it will give? this code only creates java object, no heavy compute is there

Reply:
As per my understanding.
1. For normal hybrid search without concurrent segment search I do not these objects to be initialized.
2. Also it is not a good practice to initialize object globally. 

@navneet1v your thoughts?

Reply:
1. is not a benefit, initialization is cheap, let's not over engineer here
2.  not sure which practice you're referring to as "bad". One potential cons I can think of - for testing we would be able to inject mock merger for some test case, but we don't have such tests now and we can change it anytime if needed. For scope of this change there is only one mode of how we're going to use both merger classes, so it's ok to init at class property level. In case that changes in future we can change the code.